pkg/endpointmanager/host.go

```diff
@@ -61,17 +63,37 @@ func (mgr *endpointManager) OnUpdateNode(oldNode, newNode *slim_corev1.Node,
 	oldNodeLabels := oldNode.GetLabels()
 	newNodeLabels := newNode.GetLabels()
 
-	nodeEP := mgr.GetHostEndpoint()
-	if nodeEP == nil {
-		log.Error("Host endpoint not found")
-		return nil
-	}
+	oldNodeAnnotations := oldNode.GetAnnotations()
+	newNodeAnnotations := newNode.GetAnnotations()
 
-	node.SetLabels(newNodeLabels)
+	labelsEqual := comparator.MapStringEquals(oldNodeLabels, newNodeLabels)
+	annotationsEqual := comparator.MapStringEquals(oldNodeAnnotations, newNodeAnnotations)
```

Comment:
There is a factory function available to compare annotations - https://github.com/cilium/cilium/blob/main/pkg/k8s/factory_functions.go#L67. Any reason why we don't use it here? The watcher subsystem uses the same function.

Reply:
Good call, no reason here, I just stumbled upon the `comparator` pkg in the Watcher code before I actually saw that function.

Reply:
Looking at this again, I think I do prefer to use the `comparator` package here. 

1. There is no comparable `LabelsEqual` functions, so I'd be importing `comparator` anyway.
2. I don't need to filter the annotations for comparison.
3. The watcher package also uses `comparator` internally as well.

Reply:
Yes I think so, at this level, we simply want to know if any annotation has changed between the old and the new node.